src/extension.js

```diff
@@ -0,0 +1,52 @@
+const vscode = require('vscode');
+const child_process = require('child_process');
+const util = require('util');
+
+const exec = util.promisify(child_process.exec);
+
+const extensionName = "hledger";
+
+function activate(context) {
+  const accountAutocompleteProviderSubscription =
+    vscode.languages.registerCompletionItemProvider("hledger", {
+      async provideCompletionItems(
+        document,
+        position,
+        token,
+        context
+      ) {
+        const text = document.lineAt(position).text;
+        if (!text.match(/^\s/)) {
+          // Do not provide account name completions if the line is not indented.
+          return;
+        }
+
+        const configuration = vscode.workspace.getConfiguration();
+        const hledgerPath = configuration.get(
+          `${extensionName}.hledgerPath`,
+          "hledger"
+        );
+        const ledgerFile = configuration.get(`${extensionName}.ledgerFile`, "");
+
+        const env = process.env;
+        if (ledgerFile) {
+          env.LEDGER_FILE = ledgerFile;
```

Comment:
Does this override the global env? Is that OK? I wasn't sure if we should copy the env and pass to `exec`. I guess it does no harm if nothing else in the process is using `LEDGER_FILE`. Do extensions each get their own process or is this shared across all of VS Code? Sorry if these are dumb questions, I'm just thinking out loud.